test/extended/templates/templateservicebroker_e2e.go

```diff
@@ -72,6 +75,10 @@ var _ = g.Describe("[templates] templateservicebroker end-to-end test", func() {
 		})
 		o.Expect(err).NotTo(o.HaveOccurred())
 
+		// give test user access to the openshift project's templates ... now needed with provision's SAR for get on templates (previously commented out)
+		err = cli.AsAdmin().Run("policy").Args("add-role-to-user", "edit", cli.Username(), "-n", "openshift").Execute()
```

Comment:
OK, I have confirmed that this  `add-role-to-user` for the `openshift` namespace in the extended tests is needed for the SAR on gets for templates in the provision path to work..  Removing these calls, and that SAR fails.

I've added comments detailing this around this call, as the SAR in question in provision.go.

At this point, I think we should leave this as is until the service catalog changes arrive, and it is actually authenticating, and we are using the user.Info from the api.Context instead of the bind/provision request parameter map.  At that time, when all these //TODO workaround comments are cleaned up, we remove the `add-role-to-user` calls in the extended tests, and verify this SAR still passes / fails as appropriate.